File: Lab Report 2/Task1.cpp
#include<iostream>
using namespace std;

class complex {
    private:
        double re, im;
    public:
        complex();
        complex(double r, double i);
        // complex negate();       //returns complex object
        void show();
        void input();
        void addCom(complex c1, complex c2);
        void subCom(complex c1, complex c2);
        void mulCom(complex c1, complex c2);
};

complex::complex():re(0),im(0){}

complex::complex(double r, double i):re(r),im(i){}

// complex complex::negate() {
//     complex temp;
//     temp.re= - re;
//     temp.im= - im;
//     return temp;
// }

void complex::show() {
    if(im>0)
        cout << re << " + " << im << "i" << endl;
    else
        cout << re << im << "i" << endl;
}

void complex::input() {
    cout<<"\nEnter real number: ";
    cin>>re;
    cout<<"\nEnter imaginary number: ";
    cin>>im;
}

void complex::addCom(complex c1, complex c2) {
    re = c1.re + c2.re;
    im = c1.im + c2.im;    
}

void complex::subCom(complex c1, complex c2) {
    re = c1.re - c2.re;
    im = c1.im - c2.im;    
}

void complex::mulCom(complex c1, complex c2) {
    re = (c1.re * c2.re) - (c1.im * c2.im);
    im = (c1.re * c2.im) + (c2.re * c1.im);
}

main() {
    complex c1,c2,c;
    
    c1.input();
    c1.show();

    c2.input();
    c2.show();

    c.addCom(c1, c2);
    cout<<"Sum = ";
    c.show();

    c.subCom(c1, c2);
    cout<<"Difference = ";
    c.show();

    c.mulCom(c1, c2);
    cout<<"Product = ";
    c.show();
}

File: Lab Report 2/Python/Task1.py
class Complex:
    re = 0
    im = 0

    def __init__(self):
        self.re = 0
        self.im = 0

    def __init__(self, r, i):
        self.re = r
        self.im = i

    def show(self):
        if self.im >= 0:
            print(f"{self.re}+{self.im}i")
        else:
            print(self.re, self.im, "i", sep="")

    def addCom(self, c1, c2):
        self.re = c1.re + c2.re
        self.im = c1.im + c2.im

    def subCom(self, c1, c2):
        self.re = c1.re - c2.re
        self.im = c1.im - c2.im

    def mulCom(self, c1, c2):
        self.re = (c1.re * c2.re) - (c1.im * c2.im)
        self.im = (c1.re * c2.im) + (c2.re * c1.im)

    def input(self):
        self.re = int(input("Enter real number: "))
        self.im = int(input("Enter imaginary number: "))


c1 = Complex(0,0)
c2 = Complex(0,0)
c = Complex(0,0)

print("For Complex Number C1:")
c1.input()
print("C1:", end = " ")
c1.show()

print("For Complex Number C2:")
c2.input()
print("C2:", end = " ")
c2.show()

print("Sum =", end = " ")
c.addCom(c1, c2)
c.show()

print("Difference =", end = " ")
c.subCom(c1, c2)
c.show()

print("Product =", end = " ")
c.mulCom(c1, c2)
c.show()